lib/join_collection.rb

```diff
@@ -82,7 +86,7 @@ def join_data(mapper, key, delegate_if, delegate_fields)
         when plural_target
           doc[plural_target] = target_objects if join_type == :join_many
         else
-          doc["#{singular_target}_#{field}"] = target_object.try(field)
+          doc["#{singular_target}_#{field}"] = target_object.try!(field)
```

Comment:
you may refer spec/join_collection_spec.rb line:57
```
 it 'should raise no method error if the target object does not have the field' do
```
it expect for raise no method error.